src/webgpu/api/operation/sampling/anisotropy.spec.ts

```diff
@@ -6,14 +6,506 @@ Note that anisotropic filtering is never guaranteed to occur, but we might be ab
+
+g.test('anisotropic_filter_checkerboard')
+  .desc(
+    `anisotropic filter rendering tests that draws a slanted plane and samples from a texture
+    that only has a top level mipmap, the content of which is like a checkerboard.
+    We will check the rendering result using sampler with maxAnisotropy values to be
+    different from each other, as the sampling rate is different.
+    We will also check if those large maxAnisotropy values are clamped so that rendering is the
+    same as the supported upper limit say 16.
+    A smiliar webgl demo is at https://jsfiddle.net/yqnbez24`
+  )
+  // .unimplemented();
+  // .params([
+  //   ...poptions('maxAnisotropy', [1, 2, 4, 16]),
+  // ])
+  .fn(async t => {
+    t.setup();
+
+    // init texture with only a top level mipmap
+    const textureSize = 32;
+    const texture = t.device.createTexture({
+      mipLevelCount: 1,
+      size: { width: textureSize, height: textureSize, depth: 1 },
+      format: kTextureFormat,
+      usage: GPUTextureUsage.COPY_DST | GPUTextureUsage.SAMPLED,
+    });
+
+    const textureEncoder = t.device.createCommandEncoder();
+
+    const bufferSize = kBytesPerRow * textureSize; // RGBA8 for each pixel (256 > 16 * 4)
+
+    // init checkerboard texture data
+    const data: Uint8Array = new Uint8Array(bufferSize);
+    for (let r = 0; r < textureSize; r++) {
+      const o = r * kBytesPerRow;
+      for (let c = o, end = o + textureSize * 4; c < end; c += 4) {
+        const cid = (r + (c - o) / 4) % 2;
+        const color = checkerColors[cid];
+        data[c] = color[0];
+        data[c + 1] = color[1];
+        data[c + 2] = color[2];
+        data[c + 3] = color[3];
+      }
+    }
+    const buffer = t.device.createBuffer({
+      usage: GPUBufferUsage.COPY_SRC | GPUBufferUsage.COPY_DST,
+      size: data.byteLength,
+    });
+    t.device.defaultQueue.writeBuffer(buffer, 0, data);
+    const bytesPerRow = kBytesPerRow;
+    const rowsPerImage = textureSize;
+
+    textureEncoder.copyBufferToTexture(
+      {
+        buffer,
+        bytesPerRow,
+        rowsPerImage,
+      },
+      {
+        texture,
+        mipLevel: 0,
+        origin: [0, 0, 0],
+      },
+      [textureSize, textureSize, 1]
+    );
+
+    t.device.defaultQueue.submit([textureEncoder.finish()]);
+
+    const textureView = texture.createView();
+
+    const samplers = [
+      t.device.createSampler({
+        magFilter: 'linear',
+        minFilter: 'linear',
+        mipmapFilter: 'linear',
+        maxAnisotropy: 1,
+      }),
+      t.device.createSampler({
+        magFilter: 'linear',
+        minFilter: 'linear',
+        mipmapFilter: 'linear',
+        maxAnisotropy: 16,
+      }),
+      t.device.createSampler({
+        magFilter: 'linear',
+        minFilter: 'linear',
+        mipmapFilter: 'linear',
+        maxAnisotropy: 1024,
+      }),
+    ];
+
+    const byteLength = kRTSize * kBytesPerRow;
+
+    const dst: GPUBuffer[] = [];
+
+    Promise.all(
```

Comment:
I think you need to await this, or the test could end before the expectations are registered (if you didn't see this fail locally, I wonder why).

Personally I'd rephrase this to just use a regular loop instead of Promise.all, it would be easier to read with straight-line async code.
The loop could just be `for (const maxAnisotropy of [1, 16, 1024])` I think. Then you would only need one createSampler() line, as well.